railties/lib/rails/generators/rails/app/templates/config/databases/mysql.yml.tt

```diff
@@ -6,11 +6,14 @@
 # Ensure the MySQL gem is defined in your Gemfile
 #   gem "mysql2"
 #
+# Ensure the Active Record is configured to use mysql2
+#   config.active_record.mysql_adapter = :mysql2
+#
 # And be sure to use new-style password hashing:
 #   https://dev.mysql.com/doc/refman/5.7/en/password-hashing.html
 #
 default: &default
-  adapter: mysql2
+  adapter: mysql
```

Comment:
I don't think we should change the `adapter` keys in `database.yml`. I want to replace `DATABASE_URL=mysql2://` by `DATABASE_URL=mysql://` because it usually comes from the platform/infrastructure, so it shouldn't need to know about Rails adapters, only about database protocols.

But that doesn't apply to the `adapter` key IMO.

Reply:
Yeah that's fine. When you explain it this way it's just a code change in most cases.

Not necessarily, I had a deployment setup where the YAML config files were stored as encrypted assets and written on deploy (for better or worse). But it still _supports_ `adapter: mysql` but doesn't default to it, so the user can decide.